compiler/AST/iterator.cpp

```diff
@@ -108,6 +109,63 @@ removeRetSymbolAndUses(FnSymbol* fn) {
   return yieldedType;
 }
 
+
+//
+// Handle the shape of the yielded values.
+//
+
+//
+// Insert before 'ref':
+//   temp = chpl_computeIteratorShape(shapeSpec);
+//   ir._shape_ = temp;
+//
+// Add _shape_ field to ir.type and figure out its type,
+// if the field did not exist.
+//
+CallExpr* setIteratorRecordShape(Expr* ref, Symbol* ir, Symbol* shapeSpec) {
+  // We could skip this if the field already exists and is void.
+  // It might be better to insert these anyway for uniformity.
+  VarSymbol* value  = newTemp("shapeTemp");
+  CallExpr* compute = new CallExpr("chpl_computeIteratorShape", shapeSpec);
+  CallExpr* set     = new CallExpr(PRIM_MOVE, value, compute);
+  ref->insertBefore(new DefExpr(value));
+  ref->insertBefore(set);
+  resolveExpr(compute);
+  resolveExpr(set);
+
+  AggregateType* iRecord = toAggregateType(ir->type);
+  INT_ASSERT(iRecord->symbol->hasFlag(FLAG_ITERATOR_RECORD));
+  Symbol* field = iRecord->getField("_shape_", false);
+  if (field == NULL) {
+    field = new VarSymbol("_shape_", value->type);
+    iRecord->fields.insertAtTail(new DefExpr(field));
+    // TODO: what should accessor's return intent be?
```

Comment:
`const ref`, right?

Reply:
I'm a little surprised you bothered to run build_accessor just to call it. Why not just use PRIM_GET_MEMBER_VALUE say?

Reply:
> `// TODO: what should accessor's return intent be?`

If the field stores a class (under BaseDom), as is the case in this PR, it is simpler to return it by value, rather than by const ref, as you suggest. However, it is not a big deal, and we may store something else, not a class, in the future. So I will remove the comment.

Reply:
> I'm a little surprised you bothered to run build_accessor

It is convenient to get `_shape_` from Chapel code. See the module changes in this PR.

To get `_shape_` in Chapel, gotta have an accessor.

Reply:
That seems reasonable, then. Might be worth putting a comment to why we are building the accessor.